lib/puppet/provider/rabbitmq_exchange/rabbitmqadmin.rb

```diff
@@ -34,11 +32,11 @@ def self.instances
           name = ''
         end
         # Convert output of arguments from the rabbitmqctl command to a json string.
-        if !arguments.nil?
+        if arguments.nil?
+          arguments = '{}'
+        else
           arguments = arguments.gsub(%r{^\[(.*)\]$}, '').gsub(%r{\{("(?:.|\\")*?"),}, '{\1:').gsub(%r{\},\{}, ',')
```

Comment:
@root-expert I know I had one case where the frozen string literals caused issues with reassigning a variable like this...I don't think it was in this case, but this one will be Ok, right?

Reply:
`String#gsub` return a new String so this should be safe. `String#gsub!` would not.